OsmAnd/src/net/osmand/plus/routing/data/AnnounceTimeDistances.java

```diff
@@ -138,7 +138,16 @@ public int getImminentTurnStatus(float dist, Location loc) {
 	public boolean isTurnStateActive(float currentSpeed, double dist, int turnType) {
 		switch (turnType) {
 			case STATE_TURN_NOW:
-				return isDistanceLess(currentSpeed, dist, TURN_NOW_DISTANCE, TURN_NOW_TIME);
+				if (dist <= currentSpeed * TURN_NOW_TIME) {
+					return true;
+				}
+				int leadDist = TURN_NOW_DISTANCE;
+				// Issue #17376: low speed adjustment for TURN_NOW timing:
+				if (currentSpeed > 0 && currentSpeed < DEFAULT_SPEED) {
+					float scaleDown = (float) (currentSpeed / DEFAULT_SPEED);
```

Comment:
hm.. here you'vot something obvious
newLeadDist = currentSpeed / DEFAULT_SPEED * leadDist
Math.max(newLeadDist, currentSpeed / DEFAULT_SPEED * newLeadDist)
// but you don't need it cause your currentSpeed < DEFAULT_SPEED - so no need to call it

return dist <= Math.max(POSITIONING_TOLERANCE, scaleDown * leadDist);

Reply:
I see it even 1 liner:
```
if (currentSpeed > 0 && currentSpeed < DEFAULT_SPEED) {
   return dist <= Math.max(POSITIONING_TOLERANCE, currentSpeed / DEFAULT_SPEED * TURN_NOW_DISTANCE);
}
```

Reply:
Yes, 1-liner is fine - done. ;)

Reply:
Where?